### PR TITLE
Ios content api

### PR DESCRIPTION
**Pull request by arcturus**
@wilsonpage r?

### CODE REVIEW

File: ios/ApiChannels.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    
+    let url = NSURL(string: ApiChannels.URL_STRING)
+    
+    let request = NSURLRequest(URL: url!, cachePolicy: NSURLRequestCachePolicy.ReloadIgnoringLocalCacheData, timeoutInterval: 30)
+    
+    Alamofire.request(request).responseJSON { response in
+      guard (response.result.value != nil) else {
+        callback.onError("Error performing request \(ApiChannels.URL_STRING)")
+        return
+      }
```
Comment (wilsonpage) [inline ios/ApiChannels.swift:31]: NIT: Linebreak after block

File: ios/ApiMagnetReact.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+//
+
+import Foundation
+
+@objc(ApiMagnetReact) class ApiMagnetReact: NSObject {
+  private let api: ApiMagnet = ApiMagnet()
+  
+  @objc func get(path: String,
+                 resolve: RCTPromiseResolveBlock,
+                 reject: RCTPromiseRejectBlock) {
```
Comment (wilsonpage) [inline ios/ApiMagnetReact.swift]: NIT: Don't think we use this argument formating style anywhere else in the swift code?

File: ios/Api.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import Foundation
+import SwiftyJSON
+
+protocol Api {
+  func get(path: String, callback: ApiCallback)
+  func post(path: String, data: NSDictionary, callback: ApiCallback)
+  func put(path: String, data: NSDictionary, callback: ApiCallback)
+  func delete(path: String, data: NSDictionary, callback: ApiCallback)
+  func mount(path: String, api: Api)
+}
```
Comment (wilsonpage) [inline ios/Api.swift:18]: Do we need this `protocol`?

File: ios/ApiMagnetReact.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    }))
+  }
+  
+  @objc func post(path: String,
+                  data: NSDictionary,
+                  resolve: RCTPromiseResolveBlock,
+                  reject: RCTPromiseRejectBlock) {
+    
+    api.post(path, data: data, callback: ApiCallback(success: { (result) in
+      resolve(result.rawValue)
```
Comment (wilsonpage) [inline ios/ApiMagnetReact.swift]: Indentation

File: ios/ApiMagnetReact.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        reject("get_error", "Error resolving \(path)", err)
+    }))
+  }
+  
+  @objc func post(path: String,
+                  data: NSDictionary,
+                  resolve: RCTPromiseResolveBlock,
+                  reject: RCTPromiseRejectBlock) {
+    
+    api.post(path, data: data, callback: ApiCallback(success: { (result) in
```
Comment (wilsonpage) [inline ios/ApiMagnetReact.swift]: Previously didn't use braces on argument. Can we be consistent?

File: ios/ApiMagnetReact.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    }))
+  }
+  
+  @objc func put(path: String,
+                 data: NSDictionary,
+                 resolve: RCTPromiseResolveBlock,
+                 reject: RCTPromiseRejectBlock) {
+    
+    api.put(path, data: data, callback: ApiCallback(success: { (result) in
+      resolve(result.rawValue)
```
Comment (wilsonpage) [inline ios/ApiMagnetReact.swift]: Indentation

File: ios/ApiMagnetReact.swift [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }))
+  }
+  
+  @objc func delete(path: String,
+                    data: NSDictionary,
+                    resolve: RCTPromiseResolveBlock,
+                    reject: RCTPromiseRejectBlock) {
+    
+    api.delete(path, data: data, callback: ApiCallback(success: { (result) in
+      resolve(result.rawValue)
```
Comment (wilsonpage) [inline ios/ApiMagnetReact.swift]: Indentation

File: ios/Cartfile.resolved [hunk 8]
```diff
@@ -1,4 +1,4 @@
 github "Alamofire/Alamofire" "3.5.1"
 github "stephencelis/SQLite.swift" "8b95a0c4a884f35e5102cd306f634680255473bb"
 github "SwiftyJSON/SwiftyJSON" "54017d514a87b2b855b407131292c510cdeb65f8"
-github "mozilla-magnet/magnet-scanner-ios" "0.3.2"
+github "mozilla-magnet/magnet-scanner-ios" "0.3.3"
```
Comment (wilsonpage) [inline ios/Cartfile.resolved:4]: What changes does this pull in?

File: ios/Magnet/AppDelegate.m [hunk 9]
```diff
@@ -22,9 +22,9 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   NSURL *jsCodeLocation;
   
   // Load from the server
-  //jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];
+  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];
   // Load from local bundle
-  jsCodeLocation = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+  //jsCodeLocation = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
```
Comment (wilsonpage) [inline ios/Magnet/AppDelegate.m]: Did you mean to change this?

File: ios/RequestStoreSQLite.swift [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  }
+  
+  func delete(_url: String) {
+    let item = requestStore.filter(url == _url)
+    try! db.run(item.delete())
+  }
+  
+  func get(_url: String) -> JSON? {
+    let query = requestStore.filter(url == _url).limit(1)
+
```
Comment (wilsonpage) [inline ios/RequestStoreSQLite.swift]: NIT: Unnecessary line break

File: ios/ApiSubscriptions.swift [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+//  Created by Francisco Jordano on 02/11/2016.
+//  Copyright © 2016 Mozilla. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
+
+class ApiSubscriptions: ApiBase {
+  
+  private static let PATH = "content://subscriptions"
```
Comment (wilsonpage) [inline ios/ApiSubscriptions.swift:14]: Unused var?

File: ios/System.swift [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+//  Copyright © 2016 Mozilla. All rights reserved.
+//
+
+import Foundation
+
+
+import SystemConfiguration
+
+class System {
+  class func connectedToNetwork() -> Bool {
```
Comment (wilsonpage) [inline ios/System.swift]: Are you sure you can't use the `static` keyword here? A lot more readable.

File: ios/ApiChannels.swift (hunk 1 above)
Comment (arcturus) [reply ios/ApiChannels.swift:31]: Sure

File: ios/ApiMagnetReact.swift (hunk 2 above)
Comment (arcturus) [reply ios/ApiMagnetReact.swift]: Sure, with the longer names in the previous signature was a pain, but now we can go back to normal formatting

File: ios/Api.swift (hunk 3 above)
Comment (arcturus) [reply ios/Api.swift:18]: Unfortunately in swift there is no abstract classes, but the protocol is the interface. Then we use the ApiBase as a superclass for the rest of the apis.

Strictly not necessary but like to keep it.

File: ios/Magnet/AppDelegate.m (hunk 9 above)
Comment (arcturus) [reply ios/Magnet/AppDelegate.m]: Definitely not on purpose, good catch!

File: ios/System.swift (hunk 12 above)
Comment (arcturus) [reply ios/System.swift]: we can use static there, that code is actually taken from stackoverflow and i don't think we will need to overwrite it. The difference between static and class in static methods is that if you have static you won't be able to over write in a subclass.

File: ios/ApiSubscriptions.swift (hunk 11 above)
Comment (arcturus) [reply ios/ApiSubscriptions.swift:14]: yup, from previous implementation

**Pull request merged**